flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/LegacyHeapOptions.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util.config.memory;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.MemorySize;
+
+/**
+ * Legacy JVM heap/process memory options to fallback to new options.
+ */
+public class LegacyHeapOptions {
```

Comment:
How about name this `ProcessMemoryFallbackOptions` or `ProcessMemoryLegacyOptions`?
I think "heap" could be ambiguous because it's used both in the old configuration as the total memory, and in the new configuration as the java heap memory. To that end, "process" might be a clearer phrase.

Reply:
True, I would suggest then just `LegacyMemoryOptions` with more details in jdoc comment.